traits/trait_list_object.py

```diff
@@ -53,465 +90,769 @@ def __init__(self, index=0, removed=None, added=None):
         self.added = added
 
 
-class TraitListObject(list):
-    """ A subclass of list that fires trait events when mutated.
-
-    This is used by the List trait type, and all values set into a List
-    trait will be copied into a new TraitListObject instance.
-
-    Mutation of the TraitListObject will fire a "name_items" event with
-    appropriate index, added and removed values.  In the case of setting
-    or deleting items with a slice, the index will hold either:
-
-    - the location of the first element changed, if the step is 1.  In
-      this case the last element changed can be inferred from the
-      length of the removed values.
-    - the slice that was used, otherwise.
+class TraitList(list):
+    """ A subclass of list that validates and notifies listeners of changes.
 
     Parameters
     ----------
-    trait : CTrait instance
-        The CTrait instance associated with the attribute that this list
-        has been set to.
-    object : HasTraits instance
-        The HasTraits instance that the list has been set as an attribute for.
-    name : str
-        The name of the attribute on the object.
     value : list
-        The list of values to initialize the TraitListObject with.
+        The value for the list
+    validator : callable, optional
+        Called to validate items in the list.
+        With the signature::
+
+        validator(current_list, index, removed, added)
+
+        If the validator is not provided, all values added to the list
+        are considered to be valid.
+
+    notifiers : list of callable, optional
+        A list of callables with the signature::
+
+        notifier(trait_list, index, removed, added)
+
+        If a notifier is not provided, no notification fired
```

Comment:
The idea here is to describe what happens if the creator of the `TraitList` doesn't provide the `notifiers` argument. Suggest something like "If no argument is given for *notifiers*, an empty list of notifiers is used."